internal/sirism/xml/xml.go

```
@@ -50,6 +51,80 @@ type StopMonitoringDelivery struct {
 	MonitoredStopVisitCancellations []MonitoredStopVisitCancellation `xml:"MonitoredStopVisitCancellation"`
 }
 
+func (smd *StopMonitoringDelivery) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+
+	var monitoringRef StopPointRef
+	monitoredStopVisits := make([]MonitoredStopVisit, 0)
+	monitoredStopVisitCancellations := make([]MonitoredStopVisitCancellation, 0)
+
+	for token, err := d.Token(); err == nil; token, err = d.Token() {
+		if err != nil {
+			return err
+		}
+		switch start := token.(type) {
+		case xml.StartElement:
```

Maybe log and error and continue for cases other than `xml.StartElement`. 
So that badly formed xml wil trigger a warning, instead of being silently accepted